drivers/led/aw20216s.c

```diff
@@ -0,0 +1,309 @@
+		.count = ARRAY_SIZE(transmit_spi_bufs),
+	};
+
+	// Point first part of read to null buffer so rx_buf only
+	// contains the data we want
+	struct spi_buf receive_spi_bufs[] = {
+		{
+			.buf = NULL,
+			.len = sizeof(read_cmd),
+		},
+		{
+			.buf = rx_buf,
+			.len = len,
+		},
+	};
+
+	struct spi_buf_set receive_spi_buf_set = {
+		.buffers = receive_spi_bufs,
+		.count = ARRAY_SIZE(receive_spi_bufs),
+	};
+
+	return spi_transceive_dt(&config->spi, &transmit_spi_buf_set, &receive_spi_buf_set);
+}
+
+static int aw20216s_led_set_brightness(const struct device *dev, uint32_t led, uint8_t value)
+{
+	struct aw20216s_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	if (value < dev_data->min_brightness || value > dev_data->max_brightness) {
+		return -EINVAL;
+	}
+
+	/* Set the LED brightness value */
+	uint8_t val = (value * 255U) / dev_data->max_brightness;
+
+	int err = aw20216s_write_register(dev, AW20216S_PWM_CONFIGURATION_REGISTER_BASE + led, AW20216S_PAGE_1, val);
+	if (err) {
+		LOG_ERR("Failed to set PWM configuration register %d", led);
+		return err;
+	}
+
+	return 0;
+}
+
+static inline int aw20216s_led_on(const struct device *dev, uint32_t led)
+{
+	struct aw20216s_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	if (led > AW20216S_NUM_PWM_CONFIG_REGISTERS) {
+		return -EINVAL;
+	}
+
+	LOG_DBG("LED %d on", led);
+
+	int err = aw20216s_write_register(dev, AW20216S_PWM_CONFIGURATION_REGISTER_BASE + led, AW20216S_PAGE_1, dev_data->max_brightness);
+	if (err) {
+		LOG_ERR("Failed to enable LED %d", led);
+	}
+
+	return 0;
+}
+
+static inline int aw20216s_led_off(const struct device *dev, uint32_t led)
+{
+	if (led > AW20216S_NUM_PWM_CONFIG_REGISTERS) {
+		return -EINVAL;
+	}
+
+	LOG_DBG("LED %d off", led);
+
+	int err = aw20216s_write_register(dev, AW20216S_PWM_CONFIGURATION_REGISTER_BASE + led, AW20216S_PAGE_1, 0);
+	if (err) {
+		LOG_ERR("Failed to disable LED %d", led);
+	}
+
+	return 0;
+}
+
+static int aw20216s_led_init(const struct device *dev)
+{
+	const struct aw20216s_config *config = dev->config;
+	struct aw20216s_data *data = dev->data;
+	struct led_data *dev_data = &data->dev_data;
+
+	int err;
+
+#if DT_INST_NODE_HAS_PROP(0, en_gpios)
+	if (!device_is_ready(config->enable.port)) {
+		LOG_ERR("Enable GPIO port %s not ready", config->enable.port->name);
+		return -ENODEV;
+	}
+	gpio_pin_configure_dt(&config->enable, GPIO_OUTPUT);
+	gpio_pin_set_dt(&config->enable, 1);
+	k_msleep(2);
+#endif
+
+	if (!device_is_ready(config->spi.bus)) {
+		LOG_ERR("SPI bus is not ready");
+		return -ENODEV;
+	}
+
+	dev_data->min_brightness = AW20216S_MIN_BRIGHTNESS;
+	dev_data->max_brightness = AW20216S_MAX_BRIGHTNESS;
+
+	uint8_t reset_value = 0xAE;
+	err = aw20216s_write_register(dev, AW20216S_RESET_REGISTER, AW20216S_PAGE_0, reset_value);
+	if (err) {
+		LOG_ERR("Failed to reset AW20216S");
+		return err;
+	}
+	k_msleep(2);
+
+	uint8_t default_config = 0xA0 | BIT(0);
```

Comment:
Define for magic number. But why does this need to be here and shouldn't this be provided via a binding property?